lib/jekyll/commands/serve.rb

```diff
@@ -99,9 +103,9 @@ def webrick_opts(opts)
         # Recreate NondisclosureName under utf-8 circumstance
 
         private
-        def file_handler_opts
+        def file_handler_opts(opts)
           WEBrick::Config::FileHandler.merge({
-            :FancyIndexing     => true,
+            :FancyIndexing     => opts['show_dir_listing'] ? true : false,
```

Comment:
Shouldn't this be flipped? Only do fancy indexing if `show_dir_listing` is false?